JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -86,22 +85,16 @@ public SqsNotificationListener WithMessageProcessingStrategy(IMessageProcessingS
             _handlerMap.Add(typeof(T), handlerFunc);
         }
 
-        public void Listen()
+        public void Listen(CancellationToken cancellationToken = default)
         {
             var queue = _queue.QueueName;
             var region = _queue.Region.SystemName;
             var queueInfo = $"Queue: {queue}, Region: {region}";
 
-            _cts = new CancellationTokenSource();
-            Task.Factory.StartNew(async () =>
-                {
-                    while (!_cts.IsCancellationRequested)
-                    {
-                        await ListenLoop(_cts.Token).ConfigureAwait(false);
-                    }
-                })
+            Task.Factory.StartNew(async () => { await ListenLoop(cancellationToken).ConfigureAwait(false); },
+                    cancellationToken)
                 .Unwrap()
-                .ContinueWith(t => LogTaskEndState(t, queueInfo, _log));
+                .ContinueWith(t => LogTaskEndState(t, queueInfo, _log), cancellationToken);
```

Comment:
Will this even fire if the cancellation was requested? Perhaps it should happen inside `ListenLoop` instead, just after checking that cancellation was requested?

Reply:
If can remove cancellation token here and we'll fix this bit of code with a separate PR & discussion.